code/WorkInProgress/AzrunStuff.dm

```diff
@@ -91,13 +91,19 @@
 					break
 
 			if(reagents_present)
-				SPAWN_DBG(0)
-					if(generator.active_form)
-						generator.active_form.on_revert()
-					generator.active_form = new T.type
-					generator.active_form.on_transform(generator)
+				transform_to_type(T.type)
 				return
 
+	proc/transform_to_type(type, mat_id)
```

Comment:
Can we get all this general code split into files within a code/modules/power or something

Reply:
Moved everything out to `/datums` as it isn't inherently related to power. Got rid of `AzrunStuff.dm`